core/src/sumeragi/message.rs

```diff
@@ -110,11 +130,21 @@ impl From<VersionedValidBlock> for BlockSigned {
 #[non_exhaustive]
 pub struct BlockCommitted {
     /// The corresponding block.
-    pub block: VersionedValidBlock,
+    pub block: VersionedCandidateBlock,
```

Comment:
It seems we can make this field private and add an impl like this:
```rust
impl From<BlockCommitted> for VersionedCandidateBlock {
    fn from(block: BlockCommitted) -> Self {
        Self { block: block.block }
    }
}
```
or add the `inner()` method

```rust
impl BlockCommitted {
// ...
    fn inner(self) -> VersionedCandidateBlock {
        self.block
    }
}
```

Reply:
I have plans to change `BlockCommitted` message to have only hash and signatures, so not sure if we should add this change. 
What benifits suggested approach have compared to plain field access.

Reply:
> What benefits suggested approach have compared to plain field access?

In my opinion, this approach encapsulates the inner field

Reply:
I will try to explain my point of view on this.
Imo this structure is not some kind of class where we need to held invariants it's just plain data which we use to transfer data through network, so there is no point in hiding any of its fields.

Reply:
I proceeded from the [use cases](https://github.com/hyperledger/iroha/blob/iroha2-dev/core/src/sumeragi/main_loop.rs#L718) of the field in Sumeragi. In all cases, the block is moved to a separate `VersionedValidBlock` variable, and the `BlockCommitted` is unwrapped and removed.

But yes, I understand your point. Let's leave it.